sssd_test_framework/topology_controllers.py

```diff
@@ -92,6 +93,35 @@ def topology_setup(self, client: ClientHost, ipa: IPAHost) -> None:
         super().topology_setup()
 
 
+class IPA2ClientsTopologyController(ProvisionedBackupTopologyController):
+    """
+    IPA with two clients Topology Controller.
+    """
+
+    @BackupTopologyController.restore_vanilla_on_error
+    def topology_setup(self, client: ClientHost, client2: ClientHost, ipa: IPAHost) -> None:
+        if self.provisioned:
+            self.logger.info(f"Topology '{self.name}' is already provisioned")
+            return
+
+        for cl in (client, client2):
+            self.logger.info(f"Enrolling {cl.hostname} into {ipa.domain}")
+
+            # Remove any existing Kerberos configuration and keytab
+            cl.fs.rm("/etc/krb5.conf")
+            cl.fs.rm("/etc/krb5.keytab")
+
+            # Backup ipa-client-install files
+            cl.fs.backup("/etc/ipa")
+            cl.fs.backup("/var/lib/ipa-client")
+
+            # Join ipa domain
+            cl.conn.exec(["realm", "join", ipa.domain], input=ipa.adminpw)
```

Comment:
Doesn't this already happen in https://github.com/SSSD/sssd-ci-containers/blob/master/src/ansible/roles/client/tasks/enroll_IPA.yml ?

Reply:
I think @justin-stephenson is right, and if it doesn't work. Let's make it work that way. Provisioning the client role twice, I hope, wouldn't require customization.